### DISCUSSION TITLE
Vercel ignoring poetry exported requirements.txt

### DISCUSSION THREAD

**Discussion by stabldev** [vercel/community#6848] [Help]
### Summary

Im trying to deploy a FastAPI application on Vercel with Serverless Functions using the Python Runtime.
And Im use [poetry](https://python-poetry.org/) as package manager for python.

Since vercel-python doesnt supports poetry yet, it looks for `requirements.txt` at root dir.
Example `requirements.txt` format:
```python
annotated-types==0.6.0
anyio==4.3.0
black==24.4.2
certifi==2024.2.2
```
Vercel recognize this format and everything runs smoothly with successful deployment.
At the same time `poetry` can export `requirements.txt` from its virtual environment with the command:
```bash
poetry export -f requirements.txt --without-hashes --output requirements.txt
```
This export `requirements.txt` but in different format:
```python
annotated-types==0.6.0 ; python_version >= "3.12" and python_version < "4.0"
anyio==4.3.0 ; python_version >= "3.12" and python_version < "4.0"
black==24.4.2 ; python_version >= "3.12" and python_version < "4.0"
certifi==2024.2.2 ; python_version >= "3.12" and python_version < "4.0"
cfgv==3.4.0 ; python_version >= "3.12" and python_version < "4.0"
```

If I try to deploy with this `requirements.txt` format, vercel doesnt recognize it, so it install nothing.
ps: I get this error: Code: FUNCTION_INVOCATION_FAILED

But if I try to install in my system from this format:
```bash
pip install -r requirements.txt
```
It works!!!
I tried exporting with hashes and without, yet no hope :\

my `vercel.json`:
```json
{
	"version": 2,
	"devCommand": "python3 main.py",
	"env": {
	    "PYTHONPATH": "app/"
	},
  	"builds": [
	    {
	    	"src": "/app/main.py",
	    	"use": "@vercel/python",
	    	"config": {
	    		"maxLambdaSize": "15mb",
	    		"runtime": "python3.12"
	    	}
	    }
  	],
  	"routes": [
	    {
	    	"src": "/(.*)",
	    	"dest": "app/main.py"
	    }
  	]
}
```

working repo link: https://github.com/tokitouq/mangareader-api

### Example

https://mangareader-h3tmewgmq-tokitos-projects.vercel.app/

### Steps to Reproduce

1. clone repo; https://github.com/tokitouq/mangareader-api
2. run `poetry install`
3. run `poetry export -f requirements.txt --without-hashes --output requirements.txt`
4. configure `vercel.json` and try to deploy

**Comment by RaflyLesmana3003** [comment]
been tried your example step to reproduce this issue, and i think you already solve this issue using hardcoded requirement.txt file. can we close this discussion i think?

[a comment by an automated account is not shown]
